test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,162 @@
+        } finally {
+            EventQueue.invokeAndWait(ScreenCaptureRobotTest::disposeFrame);
+        }
+    }
+
+    private static void initializeGUI() {
+        frame = new Frame("Test frame");
+        realImage = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                .getDefaultScreenDevice().getDefaultConfiguration()
+                .createCompatibleImage(200, 100);
+
+        Graphics g = realImage.createGraphics();
+        g.setColor(Color.yellow);
+        g.fillRect(0, 0, 200, 100);
+        g.setColor(Color.red);
+        g.setFont(new Font("SansSerif", Font.BOLD, 30));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+        displayImage = realImage;
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocation(100, 100);
+        frame.setVisible(true);
+        canvas.requestFocus();
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot;
+        robot = new Robot();
+        Thread.sleep(delay);
+
+        Point pnt = canvas.getLocationOnScreen();
+        Rectangle rect = new Rectangle(pnt.x + 10, pnt.y + 10, 200, 100);
+
+        // Capturing Image using Robot
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the actual image by " + difference + " pixels";
+            System.err.println("Test failed");
+            throw new RuntimeException(errorMessage);
+        }
+        System.out.println("Test passed");
+    }
+
+    private static boolean compareImages(BufferedImage capturedImg,
+            BufferedImage realImg) throws InterruptedException {
+        int[] capturedPixels;
+        int[] realPixels;
+        int imgWidth;
+        int imgHeight;
+        int toleranceLevel = 0;
+        boolean result = true;
+
+        imgWidth = capturedImg.getWidth(null);
+        imgHeight = capturedImg.getHeight(null);
+
+        capturedPixels = new int[imgWidth * imgHeight];
+        realPixels = new int[imgWidth * imgHeight];
+
+        // Extracting Pixels from Captured Image
+        PixelGrabber pgCapturedImg = new PixelGrabber(capturedImg, 0, 0,
+                imgWidth, imgHeight, capturedPixels, 0, imgWidth);
+        pgCapturedImg.grabPixels();
+        // Extracting Pixels from Real Image
+        PixelGrabber pgRealImg = new PixelGrabber(realImg, 0, 0, imgWidth,
+                imgHeight, realPixels, 0, imgWidth);
+        pgRealImg.grabPixels();
+
+        for (int i = 0; i < (imgWidth * imgHeight); i++) {
+            if (capturedPixels[i] != realPixels[i]) {
+                toleranceLevel++;
+            }
+        }
+
+        difference = toleranceLevel;
+        if (toleranceLevel > 100) {
```

Comment:
Why is a tolerance level needed in this test ?
That would seem to suggest it can't test the only thing it exists for.

Reply:
We may be comparing around 20K pixels.
Thought will keep a  0.5% variation as permissible.
Reduced a bit more now.
If this doesn’t make sense, surely we can take tolerance out.

Reply:
it should be gone. If the system is configured properly it should be a perfect match.
You probably should use -Dsun.java2d.uiScale=1 on this test to avoid any issues when run on a scaled desktop.

Reply:
Agree, Removed tolerance now.